examples/src/cpu/build_predefined_detectors.cpp

```diff
@@ -0,0 +1,162 @@
+///
+/// Telescope detector: Array of surfaces of a given type in a bounding portal
+///                     box.
+int main() {
+
+    // Memory resource to allocate the detector data stores
+    vecmem::host_memory_resource host_mr;
+
+    //
+    // Toy detector
+    //
+
+    // Toy detector type
+    using toy_detector_t = detray::detector<detray::toy_metadata<>>;
+
+    // Number of barrel layers (0 - 4)
+    constexpr std::size_t n_brl_layers{4};
+    // Number of endcap layers on either side (0 - 7)
+    // Note: The detector must be configured with 4 barrel layer to be able to
+    // add encap layers
+    constexpr std::size_t n_edc_layers{3};
+    // Fill the detector
+    const toy_detector_t toy_det =
+        detray::create_toy_geometry(host_mr, n_brl_layers, n_edc_layers);
+
+    // Print the volume graph of the toy detector
+    std::cout << "\nToy detector:\n"
+              << "-------------\n"
+              << detray::volume_graph{toy_det}.to_string() << std::endl;
+
+    //
+    // Telescope detector
+    //
+
+    // The telescope detector is built according to a 'pilot trajectory'
+    // (either a helix or a ray) along which the modules are placed in given
+    // distances. the world portals are constructed from a bounding box around
+    // the test surfaces (the envelope is configurable)
+
+    // Telescope detector tyoes containing the given module shape (can be any)
+    using rectgl_telescope_t =
+        detray::detector<detray::telescope_metadata<detray::rectangle2D<>>>;
+    using trapzd_telescope_t =
+        detray::detector<detray::telescope_metadata<detray::trapezoid2D<>>>;
+
+    // Mask with a rectangular shape (20x20 mm)
+    detray::mask<detray::rectangle2D<>> rectangle{
+        0u, 20.f * detray::unit<detray::scalar>::mm,
+        20.f * detray::unit<detray::scalar>::mm};
+
+    // Mask with a trapezoid shape
+    constexpr detray::scalar hx_min_y{10.f * detray::unit<detray::scalar>::mm};
+    constexpr detray::scalar hx_max_y{30.f * detray::unit<detray::scalar>::mm};
+    constexpr detray::scalar hy{20.f * detray::unit<detray::scalar>::mm};
+    constexpr detray::scalar divisor{10.f / (20.f * hy)};
+    detray::mask<detray::trapezoid2D<>> trapezoid{0u, hx_min_y, hx_max_y, hy,
+                                                  divisor};
+
+    // Build from given module positions (places 11 surfaces)
+    std::vector<detray::scalar> positions = {
+        0.f * detray::unit<detray::scalar>::mm,
+        50.f * detray::unit<detray::scalar>::mm,
+        100.f * detray::unit<detray::scalar>::mm,
+        150.f * detray::unit<detray::scalar>::mm,
+        200.f * detray::unit<detray::scalar>::mm,
+        250.f * detray::unit<detray::scalar>::mm,
+        300.f * detray::unit<detray::scalar>::mm,
+        350.f * detray::unit<detray::scalar>::mm,
+        400.f * detray::unit<detray::scalar>::mm,
+        450.f * detray::unit<detray::scalar>::mm,
+        500.f * detray::unit<detray::scalar>::mm};
+
```

Comment:
Note: The telescope detector will get a configuration type soon, making this a bit easier on the eyes